src/Forte.Optimizely.ContentUsage/Api/Features/ContentType/ContentTypDtoBuilder.cs

```diff
@@ -0,0 +1,42 @@
+﻿using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using EPiServer.DataAbstraction;
+
+namespace Forte.Optimizely.ContentUsage.Api.Features.ContentType;
+
+public class ContentTypeDtoBuilder
+{
+    private readonly ContentTypeUsagesRepository _contentTypeUsagesRepository;
+
+    public ContentTypeDtoBuilder(ContentTypeUsagesRepository contentTypeUsagesRepository)
+    {
+        _contentTypeUsagesRepository = contentTypeUsagesRepository;
+    }
+
+    public async Task<ContentTypeDto> Build(EPiServer.DataAbstraction.ContentType contentType,
+        CancellationToken cancellationToken)
+    {
+        var dto = new ContentTypeDto
+        {
+            ID = contentType.ID,
+            DisplayName = contentType.DisplayName,
+            Name = contentType.Name,
+            Guid = contentType.GUID,
+            Type = contentType.Base.ToString(),
+            Statistics = new UsageStatisticDto[] { }
+        };
+
+        if (contentType is not BlockType blockType)
+            return dto;
+
+        var usagePages = await _contentTypeUsagesRepository.GetUsagePages(blockType, cancellationToken);
```

Comment:
I think that's better approach but still we do one request per n-content types.
Now I encounter sometimes DB timeouts on request on first view.

But maybe it's caused by some other jobs after start KOG solution. After few minutes it works fine.